.circleci/config.yml

```diff
@@ -0,0 +1,240 @@
+            sha256sum | \
+            awk '{print "dependencies "$1}' >> $MOVEIT2_WS/checksum.txt
+  store_build_checksum: &store_build_checksum
+    store_artifacts:
+        path: checksum.txt
+  restore_build_cache: &restore_build_cache
+    restore_cache:
+      name: Restore Build Cache
+      key: build-cache-v1-{{ arch }}-{{ .Branch }}-{{ checksum "/opt/moveit2_ws/checksum.txt" }}
+  source_build: &source_build
+    run:
+      name: Source Build
+      command: |
+        if [ -d install ]
+        then
+            echo "Skipping Source Build"
+        else
+            . $ROS_WS/install/setup.sh
+            colcon build \
+              --symlink-install \
+              --cmake-args \
+                -DCMAKE_BUILD_TYPE=$CMAKE_BUILD_TYPE \
+                -DCOVERAGE_ENABLED=$COVERAGE_ENABLED
+        fi
+  save_build_cache: &save_build_cache
+    save_cache:
+      name: Save Build Cache
+      key: build-cache-v1-{{ arch }}-{{ .Branch }}-{{ checksum "/opt/moveit2_ws/checksum.txt" }}
+      paths:
+        - /opt/moveit2_ws
+  copy_build_logs: &copy_build_logs
+    run:
+        name: Copy Build Logs
+        when: always
+        command: cp -rH log/latest_build log/build
+  store_build_logs: &store_build_logs
+    store_artifacts:
+        path: log/build
+  test_build: &test_build
+    run:
+        name: Test Build
+        command: |
+          . install/setup.sh
+          src/moveit2/tools/run_test_suite.bash
+  copy_test_logs: &copy_test_logs
+    run:
+        name: Copy Test Logs
+        when: always
+        command: cp -rH log/latest_test log/test
+  store_test_logs: &store_test_logs
+    store_artifacts:
+        path: log/test
+  report_code_coverage: &report_code_coverage
+    run:
+        name: Report Code Coverage
+        command: |
+          if [ "$COVERAGE_ENABLED" = "True" ]
+          then
+            . install/setup.sh
+            src/moveit2/tools/code_coverage_report.bash codecovio
+          else
+            echo "Skipping Code Coverage Report"
+          fi
+
+commands:
+  checkout_source:
+    description: "Checkout Source"
+    steps:
+      - *pre_checkout
+      - checkout:
+          path: src/moveit2
+      - *post_checkout
+  setup_upstream:
+    description: "Setup Upstream"
+    steps:
+      - *install_upstream
+      - *restore_upstream_cache
+      - *build_upstream
+      - *save_upstream_cache
+  build_source:
+    description: "Build Source"
+    steps:
+      - *install_dependencies
+      - *store_build_checksum
+      - *restore_build_cache
+      - *source_build
+      - *save_build_cache
+      - *copy_build_logs
+      - *store_build_logs
+  test_source:
+    description: "Test Source"
+    steps:
+      - *test_build
+      - *copy_test_logs
+      - *store_test_logs
+  report_coverage:
+    description: "Report Coverage"
+    steps:
+      - *report_code_coverage
+
+executors:
+  docker_exec:
+    docker:
+      - image: rosplanning/moveit2:master
```

Comment:
we've been using the ``moveit`` dockerhub for years:
https://hub.docker.com/r/moveit/moveit/

we also just created:
https://hub.docker.com/r/moveit/moveit2

Reply:
As long as your cool co locating docker tags for users and CI. I just thought you might want to cleanly separate release and CI repos.

Reply:
Is there a good reason we'd want to separate these? We have not for years:
![image](https://user-images.githubusercontent.com/561060/54634591-29267e00-4a48-11e9-8ef4-0052ab8b5d64.png)

Reply:
A repo on Docker Hub only has one control panel to manage all of its tags, and in the future, it may be the case that two different purpose for the same repo may conflate or interfere with each other. Do you see this `moveit/moveit2` repo for pure public release distribution or continuous integration? Separating the two may be beneficial down the line, like have a separate `ros-planning/moveit2` just for CI and `moveit/moveit2` for users.

For example, I'd think it would be nice to set the source field to the build rules build and tag a respective docker image for each new devel branch using the available regex, while not interfering with build setting for release image tags. 

![image](https://user-images.githubusercontent.com/2293573/54639129-dea60100-4a49-11e9-92ad-88a16fedcf0e.png)


We may also want our CI to re-trigger all CI images, while not needlessly churning the release tags in the same repo, or modify the docker hub build to use custom build hooks customize build args:

https://github.com/ruffsl/navigation2/tree/foo/.dockerhub/devel